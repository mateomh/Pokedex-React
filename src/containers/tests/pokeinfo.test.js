import { render, screen } from '@testing-library/react';
import { Provider } from 'react-redux';
import { createStore } from 'redux';
import renderer from 'react-test-renderer';
import rootReducer from '../../reducers/root';
import PokeInfo from '../pokeinfo';

describe('PokeInfo Component', () => {
  const mockState = {
    filter: 'All',
    pokemons: [],
    page: 50,
    currentPokemon: {
      name: 'Defaulty',
      imagesm: '',
      types: ['1', '2'],
      id: 5000,
    },
  };

  const mockStore = createStore(rootReducer, mockState);

  it(
    '1. Renders correctly',
    () => {
      render(
        <Provider store={mockStore}>
          <PokeInfo />
        </Provider>,
      );
      const PokeName = screen.getByText('Defaulty');
      expect(PokeName).toBeDefined();
    },
  );

  it(
    '2. Checks it renders correctly against a snapshot',
    () => {
      const tree = renderer
        .create(
          <Provider store={mockStore}>
            <PokeInfo />
          </Provider>,
        )
        .toJSON();
      expect(tree).toMatchSnapshot();
    },
  );
});
