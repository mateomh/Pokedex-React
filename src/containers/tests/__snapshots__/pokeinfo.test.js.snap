// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PokeInfo Component 2. Checks it renders correctly against a snapshot 1`] = `
<div
  className="Card"
>
  <div
    className="Back"
    onClick={[Function]}
  >
    &lt; Back
  </div>
  <div
    className="Name"
  >
    
    <p
      className="Number"
    >
      N°  
      5000
    </p>
    <p
      className="Name"
    >
      Defaulty
    </p>
  </div>
  <img
    alt="pokemon big"
    className="Image"
  />
  <p
    className="Info Title"
  >
    Type
  </p>
  <div
    className="Info Data"
  >
    <div
      className="1 Badge"
    >
      1
    </div>
    <div
      className="2 Badge"
    >
      2
    </div>
  </div>
  <p
    className="Info Title"
  >
    Height (m)
  </p>
  <p
    className="Info Data"
  >
    NaN
  </p>
  <p
    className="Info Title"
  >
    Weight (Kg)
  </p>
  <p
    className="Info Data"
  >
    NaN
  </p>
</div>
`;
